sdks/java/io/amazon-web-services2/src/main/java/org/apache/beam/sdk/io/aws2/options/AwsOptions.java

```diff
@@ -17,76 +17,118 @@
  */
 package org.apache.beam.sdk.io.aws2.options;
 
+import java.net.URI;
 import org.apache.beam.sdk.annotations.Experimental;
 import org.apache.beam.sdk.annotations.Experimental.Kind;
+import org.apache.beam.sdk.io.aws2.common.ClientBuilderFactory;
+import org.apache.beam.sdk.io.aws2.common.HttpClientConfiguration;
 import org.apache.beam.sdk.options.Default;
 import org.apache.beam.sdk.options.DefaultValueFactory;
 import org.apache.beam.sdk.options.Description;
 import org.apache.beam.sdk.options.PipelineOptions;
 import org.checkerframework.checker.nullness.qual.Nullable;
 import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.ContainerCredentialsProvider;
 import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.EnvironmentVariableCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.ProfileCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.SystemPropertyCredentialsProvider;
 import software.amazon.awssdk.core.exception.SdkClientException;
 import software.amazon.awssdk.http.apache.ProxyConfiguration;
+import software.amazon.awssdk.regions.Region;
 import software.amazon.awssdk.regions.providers.DefaultAwsRegionProviderChain;
-import software.amazon.awssdk.utils.AttributeMap;
+import software.amazon.awssdk.services.sts.auth.StsAssumeRoleCredentialsProvider;
 
 /**
  * Options used to configure Amazon Web Services specific options such as credentials and region.
  */
 @Experimental(Kind.SOURCE_SINK)
 public interface AwsOptions extends PipelineOptions {
 
-  /** AWS region used by the AWS client. */
-  @Description("AWS region used by the AWS client")
+  /** Region used to configure AWS service clients. */
+  @Description("Region used by AWS service clients")
   @Default.InstanceFactory(AwsRegionFactory.class)
-  String getAwsRegion();
+  Region getAwsRegion();
 
-  void setAwsRegion(String value);
+  void setAwsRegion(Region region);
 
   /** Attempt to load default region. */
-  class AwsRegionFactory implements DefaultValueFactory<@Nullable String> {
+  class AwsRegionFactory implements DefaultValueFactory<@Nullable Region> {
     @Override
     @Nullable
-    public String create(PipelineOptions options) {
+    public Region create(PipelineOptions options) {
       try {
-        return new DefaultAwsRegionProviderChain().getRegion().id();
+        return new DefaultAwsRegionProviderChain().getRegion();
       } catch (SdkClientException e) {
         return null;
       }
     }
   }
 
-  /** The AWS service endpoint used by the AWS client. */
-  @Description("AWS service endpoint used by the AWS client")
-  String getEndpoint();
+  /** Endpoint used to configure AWS service clients. */
+  @Description("Endpoint used by AWS service clients")
+  URI getEndpoint();
```

Comment:
This is a breaking change to improve type safety. Considering endpoints are primarily used for testing purposes, there seems to be very little risk.